documentation/proposals/Proposal - Multi-Backend Input.md

```diff
@@ -111,8 +131,49 @@ Threading rules for the reference implementation (if any) will be explicitly doc
 
 **INFORMATIVE TEXT:** For example, it is illegal for GLFW functions to be called anywhere except the thread `glfwInit` was called on, and it is illegal on some operating systems (such as macOS) for `glfwInit` to be called anywhere except the thread that called `main`.
 
+## Actors
+
+Some users will want to receive events in the order that they happen. Observing state does not allow us to do this if, for instance, a button was pressed and released between update calls - the state would have a net-zero change and thus we wouldn't be able to detect such events.
+
+This is solved by the actor model, which is implemented by the custom list types - the actor methods invoking the matching events.
+
+**INFORMATIVE TEXT:** A possible question is "why don't we just expose the events directly on the devices? Why do we need this actor model?". The answer to that is having events on the devices has considerations for the lifetime of the device objects, which the Silk.NET team wishes to leave unrestricted. For instance, when a device connects the input context will need to bind the events on each device itself and when a device disconnects the behaviour thereafter would be undefined because we have not explicitly allowed or disallowed the reuse of device objects, and we do not wish to. If a device object were to be reused when the device reconnects, those events would still be there. It's better to encapsulate these events in their own object using this actor model.
+
+```cs
+public interface IActor
```

Comment:
Naming: I dislike actor as name, I think its generally associated with the actor pattern/model (distributed computing, think Orleans) and ´IActor´ is bound to collide with someone

Reply:
Hmmm. `IInputReceiver`? `IInputListener`? Thoughts on an alternative name?

Reply:
`IInputHandler` might make more sense if the methods it defines also say "handler"